cap-async-std/src/fs_utf8/mod.rs

```diff
@@ -68,7 +70,14 @@ fn to_utf8<P: AsRef<std::path::Path>>(path: P) -> std::io::Result<String> {
     };
 
     #[cfg(windows)]
-    let cstr = { panic!("windows not yet implemented") };
+    let cstr = {
+        use std::{ffi::CString, os::windows::ffi::OsStrExt};
+        let utf16: Vec<_> = osstr.encode_wide().collect();
+        let str = String::from_utf16(&utf16).map_err(|_| {
+            std::io::Error::new(std::io::ErrorKind::InvalidData, "invalid path string")
+        })?;
+        CString::new(str)?
+    };
```

Comment:
This code looks good. Long term, I think arf-strings will also need a `WasiString::from_maybe_nonutf16` or so, but we can figure that out separately.